test/Main.purs

```diff
@@ -63,3 +66,8 @@ main = do
   print (toSignature (Proxy :: Proxy Foo))
   print (toSignature (Proxy :: Proxy (Array Foo)))
   print (toSignature (Proxy :: Proxy MyNewString))
+  log "Basic test of Eq GenericSignature instance:"
+  log $ "Foo == Foo:" <> show (toSignature (Proxy :: Proxy Foo) == toSignature (Proxy :: Proxy Foo))
```

Comment:
I think `purescript-assert` would be good here, and we can add it as a dev dependency in `bower.json`.